File: implementation/scipamato/scipamato-core-web/src/main/java/ch/difty/scipamato/core/web/common/CoreWebSessionFacade.java
package ch.difty.scipamato.core.web.common;

import java.util.Arrays;

import org.apache.wicket.authroles.authorization.strategies.role.Roles;
import org.springframework.stereotype.Component;

import ch.difty.scipamato.common.navigator.ItemNavigator;
import ch.difty.scipamato.common.web.ScipamatoWebSessionFacade;
import ch.difty.scipamato.core.ScipamatoSession;

@Component
public class CoreWebSessionFacade implements ScipamatoWebSessionFacade {

    private static final long serialVersionUID = 1L;

    @Override
    public String getLanguageCode() {
        return ScipamatoSession
            .get()
            .getLocale()
            .getLanguage();
    }

    @Override
    public ItemNavigator<Long> getPaperIdManager() {
        return ScipamatoSession
            .get()
            .getPaperIdManager();
    }

    @Override
    public boolean hasAtLeastOneRoleOutOf(final String... roles) {
        final Roles authorizedRoles = ScipamatoSession
            .get()
            .getRoles();
        if (authorizedRoles != null)
            return Arrays
                .stream(roles)
                .anyMatch(authorizedRoles::hasRole);
        return false;
    }

}

File: core/core-web/src/main/java/ch/difty/scipamato/core/web/common/CoreWebSessionFacade.kt
package ch.difty.scipamato.core.web.common

import ch.difty.scipamato.common.navigator.ItemNavigator
import ch.difty.scipamato.common.web.ScipamatoWebSessionFacade
import ch.difty.scipamato.core.ScipamatoSession.Companion.get
import org.springframework.stereotype.Component

@Component
class CoreWebSessionFacade : ScipamatoWebSessionFacade {

    override val languageCode: String get() = get().locale.language
    override val paperIdManager: ItemNavigator<Long> get() = get().paperIdManager

    override fun hasAtLeastOneRoleOutOf(vararg roles: String): Boolean = roles.any { get().roles.hasRole(it) }

    companion object {
        private const val serialVersionUID = 1L
    }
}
